demos/python_demos/asl_recognition_demo/asl_recognition_demo/video_library.py

```diff
@@ -74,48 +75,95 @@ def parse_source_paths(input_dir, valid_names):
 
         return out_file_paths
 
-    def release(self):
-        """Release internal storages"""
-
-        if self.cap is not None:
-            self.cap.release()
-
-        self.cap = None
-
     def next(self):
         """Moves pointer to the next video source"""
 
-        self.release()
+        with self.cur_source_id.get_lock():
+            self.cur_source_id.value += 1
+            if self.cur_source_id.value >= self.num_sources:
+                self.cur_source_id.value = 0
 
-        self.cur_source_id += 1
-        if self.cur_source_id >= self.num_sources:
-            self.cur_source_id = 0
+        with self.changed_source_flag.get_lock():
+            self.changed_source_flag.value = True
 
     def prev(self):
         """Moves pointer to the previous video source"""
 
-        self.release()
+        with self.cur_source_id.get_lock():
+            self.cur_source_id.value -= 1
+            if self.cur_source_id.value < 0:
+                self.cur_source_id.value = self.num_sources - 1
 
-        self.cur_source_id -= 1
-        if self.cur_source_id < 0:
-            self.cur_source_id = self.num_sources - 1
+        with self.changed_source_flag.get_lock():
+            self.changed_source_flag.value = True
 
-    def get_frame(self):
-        """Returns current frame from the active video source"""
+    def start(self):
+        """Starts internal threads"""
 
-        source_name, source_path = self.cur_source
+        if self._play_process is not None and self._play_process.is_alive():
+            return
 
-        if self.cap is None:
-            self.cap = cv2.VideoCapture(source_path)
+        self._play_process = \
+            Process(target=self._play,
+                    args=(self.visualizer, self.cur_source_id, self.source_paths,
```

Comment:
Is the visualizer picklable? It doesn't look like it would be, given that it stores a process handle, and if so, this won't work on Windows.

Reply:
I'm not sure that we really need to make it picklable because some other parameters (multiprocessing.Value, multiprocessing.Array) are not picklable too but it looks like it able to work on Windows?

Reply:
Have you tested this on Windows? If yes (and it works), then sure, no problem.

> some other parameters (multiprocessing.Value, multiprocessing.Array) are not picklable too

FWIW, `multiprocessing` implements special logic that allows these objects to be shared between processes.

Reply:
Ok, I'll check it.

Reply:
I've fixed demo. Now, it works on Windows too.